cpp/ps0004/150daysprogramming.cpp

```diff
@@ -1,154 +1,62 @@
-// 150daysprogramming.cpp : This file contains the 'main' function. Program execution begins and ends there.
-//
-
 #include <iostream>
-using namespace std;
-#include<string>
-
-//bool isFloat(string num);
-
-bool isFloat(string num) {
-    bool isFloating = false;
-    for (char c : num) {
-        if (c == '.') {
-            isFloating = true;
-            break;
-        }
-    }
-    return isFloating;
-}
-
-
-bool isNumber( string s)
-{
-    for (char c : s) {
-        if (s.length() > 1 && c == '-') {
-            continue;
-        }
-        if (c == '.') {
-            continue;
-        }
-        if (!isdigit(c)) {
-            return false;
-        }
-    }
-    return true;
-}
+#include <string>
 
+using namespace std;
 
+bool isNumber(string s);
+bool isFloat(string num);
 
 int main()
 {
+    // prompting user for number 1.5/1.5
     string input;
     cout << "Enter a number from 32 to 126: ";
     cin >> input;
 
-    bool isNum = isNumber(input);
-
-    if (!isNum || input=="0") {
+    // checking invalidaty of user's input 4/4
+    // printing the number and the opposite printable character 2/2
+    if (!isNumber(input) || input=="0") {
         cout << "Invalid character/s or zero value! \n";
     }
     else {
-        bool isFloating = isFloat(input);
-        if (isFloating) {
+        if (isFloat(input)) {
             cout << "Invalid: a floating point number! \n";
         }
         else {
             int intNumber = stoi(input);
             if (intNumber >= 32 && intNumber <= 126) {
 
-                cout << "The character for " << intNumber << " is "<< char(intNumber) << " .";
+                cout << "The character for " << intNumber << " is " << char(intNumber) << ".";
             }
             else {
                 cout << "Invalid: number is out of the range! \n";
             }
         }
-    }
-
-   
-    
-  
-    
-    
+    }    

```

Comment:
- [x] used `if` structure in a right way
- [x] `if` structure take in consideration all invalid cases
- [x] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character